extensions/crypto/runtime/src/main/doc/usage.adoc

```diff
@@ -0,0 +1,23 @@
+=== FIPS
+
+When running `crypto` extension on FIPS enabled system the BC-FIPS has to be utilized.
```

Comment:
I think that now after the module split, the statement "BC-FIPS has to be utilized" is too strong. I would say "any FIPS-compliant Java Security Provider (such as BC-FIPS) hast to be used instead of `bcprov`" is right as of Camel 4.7.0.

Actually I wonder, whether there is any reason (beyond backwards compatibility) to remove BC from the Crypto component in Camel? The Camel crypto module not only compiles but also its tests are passing after the removal of all BC dependencies. `SunJCE` provider might be just fine for the basic usage. Note that I have not looked whether the tests perform some real crypto-operations, but I strongly hope so. 

If we remove BC from the crypto component, then the component should keep working out of the box also on FIPS enabled systems, no?

Reply:
Interesting idea, I'll check if we can remove BC from camel-crypto. If it would be possible/approved, then definitely the statement  `any FIPS-compliant Java Security Provider (such as BC-FIPS) hast to be used instead of bcprov` would be much better.

Reply:
@ppalaga I checked the options and you are right. Camel-crypto does not require BC at all. I will propose such change in near future to the Camel. In the meantime I'll exclude the BC from camel-crypto in the CQ BOM (and I will revert it back as soon as the relevant Camel bits reaches the CQ)
I have a question what to suggest to users:
1 - In non-FIPS environment all worjs out of the box.
2 - In the FIPS environment, If user wants to use BCFIPS, following dependencies have to be added

```
                <dependency>
                    <groupId>org.bouncycastle</groupId>
                    <artifactId>bc-fips</artifactId>
                </dependency>
                <dependency>
                    <groupId>org.apache.camel.quarkus</groupId>
                    <artifactId>camel-quarkus-support-bouncycastle</artifactId>
                </dependency>
```

If another fips-compliant security provider is used, it has to be registered properly.

(I tested the scenario with BCFIPS on FIPS machine)

Does it make sense to you, to suggest  both approaches? (BCFIPS and different fips compliant provider)
(the other option is to suggest use of BCFIPS only)

Reply:
Fixed with explanation in the main thread

Reply:
> 1 - In non-FIPS environment all worjs out of the box.

I'd vote for being more explaining even for non-FIPS. We should note the that

* BC dependency was removed from crypto in Camel 3.8.0. 
* If users need algorithms or hash functions provided by BC, they need to add what? - camel-quarkus-support-bouncycastle AND bcprov? camel-quarkus-support-bouncycastle will do what? - register BC as a Security Provider and add native support (correct me if I am wrong)?
* When adding other non-default security providers to class path, users should also register them.

> 2 - In the FIPS environment

Sounds right.